packages/mds-jurisdiction/api.ts

```diff
@@ -0,0 +1,140 @@
+/*
+    Copyright 2019-2020 City of Los Angeles.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+import express from 'express'
+import {
+  pathsFor,
+  ServerError,
+  NotFoundError,
+  ValidationError,
+  ConflictError,
+  AuthorizationError
+} from '@mds-core/mds-utils'
+import { checkAccess, ApiVersionMiddleware } from '@mds-core/mds-api-server'
+import { JurisdictionService } from '@mds-core/mds-jurisdiction-service'
+import { Jurisdiction } from '@mds-core/mds-types'
+import {
+  JurisdictionApiGetJurisdictionsRequest,
+  JurisdictionApiGetJurisdictionsResponse,
+  JurisdictionApiCreateJurisdictionRequest,
+  JurisdictionApiCreateJurisdictionResponse,
+  JurisdictionApiGetJurisdictionResponse,
+  JurisdictionApiGetJurisdictionRequest,
+  JurisdictionApiResponse,
+  JURISDICTION_API_DEFAULT_VERSION,
+  JURISDICTION_API_SUPPORTED_VERSIONS
+} from './types'
+
+const UnexpectedServiceError = (error: ServerError | null) =>
+  error instanceof ServerError ? error : new ServerError('Unexected Service Error', { error })
+
+const HasJurisdictionClaim = <TBody extends {}>(res: JurisdictionApiResponse<TBody>) => (
+  jurisdiction: Jurisdiction
+): boolean =>
+  res.locals.scopes.includes('jurisdictions:read') ||
+  (res.locals.claims?.jurisdictions?.split(' ') ?? []).includes(jurisdiction.agency_key)
+
+function api(app: express.Express): express.Express {
+  app.use(
+    ApiVersionMiddleware(
+      'application/vnd.mds.jurisdiction+json',
+      JURISDICTION_API_SUPPORTED_VERSIONS
+    ).withDefaultVersion(JURISDICTION_API_DEFAULT_VERSION)
+  )
+  app.get(
+    pathsFor('/jurisdictions'),
+    checkAccess(scopes => scopes.includes('jurisdictions:read') || scopes.includes('jurisdictions:read:claim')),
+    async (req: JurisdictionApiGetJurisdictionsRequest, res: JurisdictionApiGetJurisdictionsResponse) => {
+      const { effective } = req.query
+
+      const [error, jurisdictions] = await JurisdictionService.getAllJurisdictions({
+        effective: effective ? Number(effective) : undefined
+      })
+
+      // Handle result
+      if (jurisdictions) {
+        return res.status(200).send({
+          version: res.locals.version,
+          jurisdictions: jurisdictions.filter(HasJurisdictionClaim(res))
+        })
+      }
+
+      // Handle errors
+      return res.status(500).send({ error: UnexpectedServiceError(error) })
```

Comment:
Why is this a 500 instead of 404? Shouldn't there be a check on `error` somewhere?

Reply:
Getting _all_ jurisdictions can't 404. `ServerError` is the only error returned by the jurisdiction service.

Reply:
In other words, it returns `200 OK` with a `[]` result.

Reply:
Oh yeah I missed that this was the _all_ endpoint.